proto/select.proto

```diff
@@ -108,13 +108,21 @@ message DAGRequest {
 
 	// Represents the maximum size of one packet, any generated string, or any parameter sent as long data.
 	optional uint64 max_allowed_packet = 13;
+
+	// Represents the tidb system endian.
+	optional Endian tidb_system_endian = 14 [(gogoproto.nullable) = false];
```

Comment:
Endian might not be enough. For example, alignment and machine width also affect the decode. I think we needs further discussion.

Reply:
I think alignment and machine width has no effect. All about alignment and machine width will be guaranteed by Golang.

Reply:
That's exactly where the problem is. Golang determines it, instead of you determines it, which means it can be architecture-dependent. For example, https://go101.org/article/memory-layout.html describes how architecture affects Golang's alignment.

Reply:
Ok. You are right.  We can add a check in tidb. If the alignment isn't what you want, use default format to encode/decode.

Reply:
How about whitelist the architecture in TiDB side for now?

Reply:
So we need whitelist the architecture, the OS ,the Go version and possible combinations? There are many combinations and we don't know the behavior.
On the one hand, only checking the alignment isn't enough? It only has influence on type Time and type Decimal, and we can check the usage of the memory in TiDB.  On the other hand, maintaining a whitelist for it is worthless. If we update the go version in the future, we also need test for the whitelist.